pkg/storage/db.go

```diff
@@ -898,6 +898,111 @@ func (db *DB) FetchByARNID(ctx context.Context, when time.Time, arnID string) ([
 	return cloudAssetDetails, err
 }
 
+// FetchByResourceID gets the assets who have resource ID at the specified time
+func (db *DB) FetchByResourceID(ctx context.Context, when time.Time, resID string) ([]domain.CloudAssetDetails, error) {
```

Comment:
Question/Suggestion: Having a separate handler mostly makes sense to me so we can have different validation messages, but a quick comparison of FetchByResourceID and FetchByARNID shows the functions are about equivalent besides having resID and arnID switched. 

Is there anything stopping you from having the handler call FetchByARNID in the background?

Reply:
I think the name "FetchByResourceID" explains what it does because the function gets resources by resource ID. Also, the name matches with handler name. Other than this, they are mostly the same.

Reply:
I think it would be fair to not copy the arnid methods and to instead rename them to resource by id which is more clear. Then have both the "arnid" endpoint and "resid" endpoint point to the same method in the backend. This is fine because the purpose of this new endpoint is for clarity to the external user. Having the same method in the backend will not affect what users expectations from these endpoints, ie they are currently supposed function the same with arnid being deprecated.

Reply:
I think this is a good suggestion. I will change the endpoints and have both endpoints point to the same method. So "fetch by arn id" will still work after deprecated.

Reply:
I believe this has been addressed.

Reply:
Indeed it has.